pkg/plugins/backendplugin/instrumentation/instrumentation.go

```diff
@@ -34,29 +42,51 @@ func instrumentPluginRequest(pluginID string, endpoint string, fn func() error)
 		status = "error"
 	}
 
-	elapsed := time.Since(start) / time.Millisecond
-	pluginRequestDuration.WithLabelValues(pluginID, endpoint).Observe(float64(elapsed))
-	pluginRequestCounter.WithLabelValues(pluginID, endpoint, status).Inc()
+	elapsed := time.Since(start)
+	pluginRequestDuration.WithLabelValues(pluginCtx.PluginID, endpoint).Observe(float64(elapsed / time.Millisecond))
+	pluginRequestCounter.WithLabelValues(pluginCtx.PluginID, endpoint, status).Inc()
+
+	if cfg.IsFeatureToggleEnabled(featuremgmt.FlagDatasourceLogger) {
+		logParams := []interface{}{
+			"status", status,
+			"duration", elapsed,
+			"plugin_id", pluginCtx.PluginID,
+			"endpoint", endpoint,
+		}
+
+		traceID := tracing.TraceIDFromContext(ctx, false)
+		if traceID != "" {
+			logParams = append(logParams, "traceID", traceID)
+		}
+
+		if pluginCtx.DataSourceInstanceSettings != nil {
+			logParams = append(logParams, "ds_name", pluginCtx.DataSourceInstanceSettings.Name)
+			logParams = append(logParams, "ds_uid", pluginCtx.DataSourceInstanceSettings.UID)
+			logParams = append(logParams, "ds_type", pluginCtx.DataSourceInstanceSettings.Type)
```

Comment:
You're mixing snake_case and camelCase here. Please try to be consistent. I think most of our logs uses snakeCase.

Reply:
:+1: